packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -82,6 +82,13 @@ export default class CallDiagnosticMetrics extends StatelessWebexPlugin {
   // @ts-ignore
   private callDiagnosticEventsBatcher: CallDiagnosticEventsBatcher;
   private logger: any; // to avoid adding @ts-ignore everywhere
+  // the default validator before piping an event to the batcher
+  // this function can be overridden by the user
+  public validator: (options: {
```

Comment:
Why is this in the `ts` file, and not the root `.js` metrics plugin file?

Reply:
because this is where it belongs. The validator applies to the call-diagnostic-metrics plugin

Reply:
Gotcha, note that we may need to elevate it as this area scales. (in the case that we will need to validate more payloads against schemas)